### ISSUE TITLE
Where to pass access_token

### ISSUE THREAD

**Issue by purplecones** [ChrisAlvares/node-adwords#31]
Where do I pass the `access_token` I receive from `getAccessTokenFromAuthorizationCode()`?

I keep getting `AuthorizationError.USER_PERMISSION_DENIED` when calling `campaignService.get()`.

I'm trying to establish my first API call to AdWords and pull some campaign data. Here is the code that throws the permission error.

```javascript
let auth = new AdwordsAuth({
    client_id: 'xxx',
    client_secret: 'xxx'
}, 'http://localhost:3000/adwords/auth' /** insert your redirect url here */);

app.get('/', (req, res) => {
    res.redirect(auth.generateAuthenticationUrl());
})
app.get('/adwords/auth', (req, res) => {
    auth.getAccessTokenFromAuthorizationCode(req.query.code, (error, tokens) => {
        console.log('tokens...');
        console.logFull({ error, tokens });
        //save access and especially the refresh tokens here

          let user = new AdwordsUser({
              developerToken: 'xxxx', //your adwords developerToken
              userAgent: 'xxx', //any company name
              clientCustomerId: 'xxxx', //the Adwords Account id (e.g. 123-123-123)
              client_id: 'xxxx', //this is the api console client_id
              client_secret: 'xxxx',
              refresh_token: tokens.refresh_token,
          });

          let campaignService = user.getService('CampaignService', 'v201702');
  
          let selector = {
              fields: ['Id', 'Name'],
              ordering: [{field: 'Name', sortOrder: 'ASCENDING'}],
              paging: {startIndex: 0, numberResults: AdwordsConstants.RECOMMENDED_PAGE_SIZE}
          };

          campaignService.get({ serviceSelector: selector }, (error, result) => {
              console.log(result.body);
              res.send(result)
          });
    })
});
```

**Comment by purplecones** [comment]
I successfully get a tokens object from `getAccessTokenFromAuthorizationCode()'
```javascript
{ access_token: 'xxx',
     refresh_token: 'xxx',
     token_type: 'Bearer',
     expiry_date: 1493582430404 }
```

I noticed refresh_token is not present if I OAuth2 is done and the app has already been given permissions to.

My confusion is why do I create a `AdwordsUser` user with the `refresh_token`. Isn't the `access_token` the thing that gives me permission to read data? In the background I would use the `refresh_token` to get a new `access_token` if its expired.

**Comment by purplecones** [comment]
I was able to gain access to the AdWords data with the following object.

```javascript
      let user = new AdwordsUser({
        developerToken: 'xxx', // your adwords developerToken
        userAgent: 'xxx', // any company name
        clientCustomerId: 'xxx-xxx-xxxx', // the Adwords Account id (e.g. 123-123-123)
        client_id: 'xxx',
        client_secret: 'xxx',
        refresh_token: tokens.refresh_token,
        access_token: tokens.access_token,
      });
```
but it only works if I specify the `clientCustomerId` of the AdWords account that is giving my app the permission. I currently hardcoded because I own the account but how would I get the `clientCustomerId` from a user after the OAuth ?

**Comment by purplecones** [comment]
Ended up using the `CustomerService` to get list of id's: https://developers.google.com/adwords/api/docs/reference/v201702/CustomerService.Customer

Thanks for creating this package btw!

**Comment by ChrisAlvares** [comment]
@purplecones I am going to reopen this as we need to add the `prompt:'consent'` to the authorization part of the `generateAuthenticationUrl` method.

**Comment by ChrisAlvares** [comment]
fixed in 3e457e601fa156959daf5dc1563ab63087e76345

**Comment by Nastromo** [comment]
> Ended up using the `CustomerService` to get list of id's: https://developers.google.com/adwords/api/docs/reference/v201702/CustomerService.Customer
> 
> Thanks for creating this package btw!

How do you get this AdWords ID? Can you show a code snippet?
